tests/test_wires.py

```diff
@@ -0,0 +1,253 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for :mod:`pennylane.wires`.
+"""
+import pytest
+import numpy as np
+from pennylane.wires import Wires, WireError
+
+
+class TestWires:
+    """Tests for the ``Wires`` class."""
+
+    @pytest.mark.parametrize("iterable", [Wires([0, 1, 2]),
+                                          np.array([0, 1, 2]),
+                                          [0, 1, 2],
+                                          (0, 1, 2),
+                                          range(3)
+                                          ])
+    def test_creation_from_common_iterables(self, iterable):
+        """Tests that a Wires object can be created from standard iterable inputs."""
+
+        wires = Wires(iterable)
+        assert wires.wire_list == [0, 1, 2]
+
+    @pytest.mark.parametrize("iterable", [[1, 0, 4],
+                                          ['a', 'b', 'c'],
+                                          ['a', 1, "ancilla"],
```

Comment:
Note that we allow nested lists for `Wires` (possibly needed for tensor product observables), then the ability to specify a wire as a string with length greater than 1 could be problematic (since it would have the same kind of sequencing semantics as a nested list)